vespaclient-container-plugin/src/test/java/com/yahoo/document/restapi/resource/DocumentV1ApiTest.java

```diff
@@ -0,0 +1,277 @@
+            executor.lastOperationContext().success(Optional.empty());
+            assertSameJson("{" +
+                           "  \"pathId\": \"/document/v1/space/music/number/1/two\"," +
+                           "  \"id\": \"id:space:music:n=1:two\"" +
+                           "}",
+                           response.readAll());
+            assertEquals(200, response.getStatus());
+            DocumentPut put = new DocumentPut(doc2);
+            put.setCondition(new TestAndSetCondition("test it"));
+            assertEquals(put, executor.lastOperation());
+            assertEquals(parameters(), executor.lastParameters());
+
+            // PUT with a document update payload is a document update operation.
+            response = driver.sendRequest("http://localhost/document/v1/space/music/group/a/three?create=true", PUT,
+                                          "{" +
+                                          "  \"fields\": {" +
+                                          "    \"artist\": { \"assign\": \"Lisa Ekdahl\" }" +
+                                          "  }" +
+                                          "}");
+            executor.lastOperationContext().success(Optional.empty());
+            assertSameJson("{" +
+                           "  \"pathId\": \"/document/v1/space/music/group/a/three\"," +
+                           "  \"id\": \"id:space:music:g=a:three\"" +
+                           "}",
+                           response.readAll());
+            assertEquals(200, response.getStatus());
+            DocumentUpdate update = new DocumentUpdate(manager.getDocumentType("music"), "id:space:music:g=a:three");
+            update.addFieldUpdate(FieldUpdate.createAssign(manager.getDocumentType("music").getField("artist"),
+                                                           new StringFieldValue("Lisa Ekdahl")));
+            update.setCreateIfNonExistent(true);
+            assertEquals(update, executor.lastOperation());
+            assertEquals(parameters(), executor.lastParameters());
+
+            // DELETE with full document ID is a document remove operation.
+            response = driver.sendRequest("http://localhost/document/v1/space/music/number/1/two?route=route", DELETE);
+            executor.lastOperationContext().success(Optional.empty());
+            assertSameJson("{" +
+                           "  \"pathId\": \"/document/v1/space/music/number/1/two\"," +
+                           "  \"id\": \"id:space:music:n=1:two\"" +
+                           "}",
+                           response.readAll());
+            assertEquals(200, response.getStatus());
+            assertEquals(new DocumentRemove(doc2.getId()), executor.lastOperation());
+            assertEquals(parameters().withRoute("route"), executor.lastParameters());
+        }
+    }
+
+    void assertSameJson(String expected, String actual) {
+        ByteArrayOutputStream expectedPretty = new ByteArrayOutputStream();
```

Comment:
Use the JSON assert helper in `testutils`

Reply:
Nice, thanks.